Invalid code generated when source class is not in a namespace

Consider a console app using top-level statements, for which a `namespace` cannot be declared:

```c#
using Figgle;

[EmbedFiggleFont(memberName: "Standard", fontName: "standard")]
static partial class Test
{
}
```

Produces:

![Image](https://github.com/user-attachments/assets/57bf8cc3-0bcc-4d45-a7ca-1f4c61e1ba85)

This is invalid code.

> Test.g.cs(15,10,15,10): error CS1001: Identifier expected

--- a/Figgle.Generator.Tests/EmbedFontSourceGeneratorTests.cs
+++ b/Figgle.Generator.Tests/EmbedFontSourceGeneratorTests.cs
@@ -218,6 +218,40 @@ public class EmbedFontSourceGeneratorTests : SourceGeneratorTests
 
         string generatedCode = compilation.SyntaxTrees.Last().ToString();
         Assert.Contains(
+            "namespace Test.Namespace",
+            generatedCode);
+        Assert.Contains(
+            "public static FiggleFont Foo => _fontByName.GetOrAdd(\"Foo\", _ => FiggleFontParser.ParseString(FooFontDescription, _stringPool));",
+            generatedCode);
+        Assert.Contains(
+            "private static readonly string FooFontDescription = @\"",
+            generatedCode);
+    }
+
+    [Fact]
+    public void ExternalFontInAdditionalFiles_NoNamespace_EmbedsFont()
+    {
+        string source =
+            """
+            using Figgle;
+
+            [EmbedFiggleFont("Foo", "ANSI Shadow")]
+            internal static partial class DemoUsage
+            {
+            }
+            """;
+
+        var additionalFont = ExternalFontAdditionalText.Create("ANSI Shadow.flf");
+        var optionsProvider = CreateOptionsProvider("ANSI Shadow.flf", "ANSI Shadow");
+
+        var (compilation, diagnostics) = RunGenerator(source, [additionalFont], optionsProvider);
+        Assert.Empty(diagnostics);
+
+        string generatedCode = compilation.SyntaxTrees.Last().ToString();
+        Assert.DoesNotContain(
+            "namespace",
+            generatedCode);
+        Assert.Contains(
             "public static FiggleFont Foo => _fontByName.GetOrAdd(\"Foo\", _ => FiggleFontParser.ParseString(FooFontDescription, _stringPool));",
             generatedCode);
         Assert.Contains(

--- a/Figgle.Generator/EmbedFontSourceGenerator.cs
+++ b/Figgle.Generator/EmbedFontSourceGenerator.cs
@@ -270,16 +270,20 @@ internal sealed class EmbedFontSourceGenerator : IIncrementalGenerator
 
     private string RenderSource(ITypeSymbol type, ImmutableArray<RenderSourceInfo> fontsToGenerate)
     {
-        return $$"""
+        string ns = type.ContainingNamespace.ToDisplayString(_fullyQualifiedFormat);
+
+        if (ns.Length is not 0)
+        {
+            return $$"""
             {{Header}}
 
             using System;
             using System.Collections.Concurrent;
             using Figgle;
 
-            namespace {{type.ContainingNamespace.ToDisplayString(_fullyQualifiedFormat)}}
+            namespace {{ns}}
             {
-                {{GetAccessibility(type.DeclaredAccessibility)}} static partial class {{type.Name}}
+                {{GetAccessibility(type.DeclaredAccessibility)}}static partial class {{type.Name}}
                 {
                     private static readonly ConcurrentDictionary<string, FiggleFont> _fontByName = new(StringComparer.Ordinal);
                     private static readonly StringPool _stringPool = new();
@@ -288,6 +292,25 @@ internal sealed class EmbedFontSourceGenerator : IIncrementalGenerator
             }
             
             """;
+        }
+        else
+        {
+            return $$"""
+                {{Header}}
+
+                using System;
+                using System.Collections.Concurrent;
+                using Figgle;
+
+                {{GetAccessibility(type.DeclaredAccessibility)}}static partial class {{type.Name}}
+                {
+                    private static readonly ConcurrentDictionary<string, FiggleFont> _fontByName = new(StringComparer.Ordinal);
+                    private static readonly StringPool _stringPool = new();
+                    {{RenderFonts(fontsToGenerate, indentationLevel: 1)}}
+                }
+            
+                """;
+        }
 
         static string RenderFonts(ImmutableArray<RenderSourceInfo> renderInfos, int indentationLevel)
         {
@@ -306,20 +329,20 @@ internal sealed class EmbedFontSourceGenerator : IIncrementalGenerator
 
             return builder.ToString();
         }
-    }
 
-    private static string GetAccessibility(Accessibility declaredAccessibility)
-    {
-        return declaredAccessibility switch
+        static string GetAccessibility(Accessibility declaredAccessibility)
         {
-            Accessibility.Public => "public",
-            Accessibility.Internal => "internal",
-            Accessibility.Private => "private",
-            Accessibility.Protected => "protected",
-            Accessibility.ProtectedAndInternal => "protected internal",
-            Accessibility.NotApplicable => string.Empty,
-            _ => throw new NotImplementedException($"Unexpected accessibility '{declaredAccessibility}'"),
-        };
+            return declaredAccessibility switch
+            {
+                Accessibility.Public => "public ",
+                Accessibility.Internal => "internal ",
+                Accessibility.Private => "private ",
+                Accessibility.Protected => "protected ",
+                Accessibility.ProtectedAndInternal => "protected internal ",
+                Accessibility.NotApplicable => "",
+                _ => throw new NotImplementedException($"Unexpected accessibility '{declaredAccessibility}'."),
+            };
+        }
     }
 
     private sealed record EmbedFontAttributeInfo(

--- a/Figgle.Generator/EmbedFontSourceGenerator.cs
+++ b/Figgle.Generator/EmbedFontSourceGenerator.cs
@@ -283,7 +283,7 @@ internal sealed class EmbedFontSourceGenerator : IIncrementalGenerator
 
             namespace {{ns}}
             {
-                {{GetAccessibility(type.DeclaredAccessibility)}}static partial class {{type.Name}}
+                static partial class {{type.Name}}
                 {
                     private static readonly ConcurrentDictionary<string, FiggleFont> _fontByName = new(StringComparer.Ordinal);
                     private static readonly StringPool _stringPool = new();
@@ -302,7 +302,7 @@ internal sealed class EmbedFontSourceGenerator : IIncrementalGenerator
                 using System.Collections.Concurrent;
                 using Figgle;
 
-                {{GetAccessibility(type.DeclaredAccessibility)}}static partial class {{type.Name}}
+                static partial class {{type.Name}}
                 {
                     private static readonly ConcurrentDictionary<string, FiggleFont> _fontByName = new(StringComparer.Ordinal);
                     private static readonly StringPool _stringPool = new();
@@ -328,20 +328,6 @@ internal sealed class EmbedFontSourceGenerator : IIncrementalGenerator
             }
 
             return builder.ToString();
-        }
-
-        static string GetAccessibility(Accessibility declaredAccessibility)
-        {
-            return declaredAccessibility switch
-            {
-                Accessibility.Public => "public ",
-                Accessibility.Internal => "internal ",
-                Accessibility.Private => "private ",
-                Accessibility.Protected => "protected ",
-                Accessibility.ProtectedAndInternal => "protected internal ",
-                Accessibility.NotApplicable => "",
-                _ => throw new NotImplementedException($"Unexpected accessibility '{declaredAccessibility}'."),
-            };
         }
     }
 

